### PR TITLE
cleanup: removes the docs from README to docs folder

### PR DESCRIPTION
**Pull request by yati1998**
This commit removes the installations methods of CSI-Addons
controller from the README file to deploy-controller.md inside
the docs folder. This will keep the README clean and have all
documentation at one place.

Signed-off-by: yati1998 <ypadia@redhat.com>

### CODE REVIEW

**Comment by yati1998** [comment]
This PR would require https://github.com/csi-addons/kubernetes-csi-addons/pull/123 to get merged first.

File: README.md [hunk 1]
```diff
@@ -57,60 +57,11 @@ By listing the `CSIAddonsNode` CRs, the CSI-Addons Controller knows how to
...
-NAME                                                    TYPE        CLUSTER-IP      EXTERNAL-IP   PORT(S)    AGE
-service/csi-addons-controller-manager-metrics-service   ClusterIP   172.30.153.17   <none>        8443/TCP   49s
-
-NAME                                            READY   UP-TO-DATE   AVAILABLE   AGE
-deployment.apps/csi-addons-controller-manager   1/1     1            1           49s
-
-NAME                                                       DESIRED   CURRENT   READY   AGE
-replicaset.apps/csi-addons-controller-manager-687d47b8c7   1         1         1       49s
-```
+The most preferred way to install the CSI-Addons Controller is by creating the
```
Comment (nixpanic) [inline README.md:60]: I think the preferred way is to use yaml files that are made available with a release like https://github.com/csi-addons/kubernetes-csi-addons/releases/tag/v0.3.0 ?

File: docs/deploy-controller.md [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+
+The CSI-Addons Controller can be deployed by different ways:
+
+## Installation for versioned deployments using
+
+The CSI-Addons Controller can also be installed  using the yaml files in `deploy/controller`.
+This is the recommended and easiest way to deploy the controller.
+
+```console
+cd deploy/controller
```
Comment (nixpanic) [inline docs/deploy-controller.md:11]: missing the `$` as prefix for a command

File: docs/deploy-controller.md [hunk 3]
```diff
@@ -0,0 +1,100 @@
+# Deploying the CSI-Addons Controller
+
+The CSI-Addons Controller can be deployed by different ways:
+
+## Installation for versioned deployments
+
+The CSI-Addons Controller can also be installed  using the yaml files in `deploy/controller`.
+The versioned deployment is possible with the yaml files that get generated for a release,
+like [release-v0.3.0](https://github.com/csi-addons/kubernetes-csi-addons/releases/tag/v0.3.0).
```
Comment (Madhu-1) [inline docs/deploy-controller.md:9]: mention to use the latest release `https://github.com/csi-addons/kubernetes-csi-addons/releases/latest`

File: README.md [hunk 4]
```diff
@@ -57,66 +57,18 @@ By listing the `CSIAddonsNode` CRs, the CSI-Addons Controller knows how to
...
-
-NAME                                            READY   UP-TO-DATE   AVAILABLE   AGE
-deployment.apps/csi-addons-controller-manager   1/1     1            1           49s
-
-NAME                                                       DESIRED   CURRENT   READY   AGE
-replicaset.apps/csi-addons-controller-manager-687d47b8c7   1         1         1       49s
-```
+The preferred way to install the controllers is to use yaml files that are
+made available with a release like 
+[release-v0.3.0](https://github.com/csi-addons/kubernetes-csi-addons/releases/tag/v0.3.0).
```
Comment (Madhu-1) [inline README.md:62]: same here mention to use latest release https://github.com/csi-addons/kubernetes-csi-addons/releases/latest

File: docs/deploy-controller.md [hunk 5]
```diff
@@ -0,0 +1,100 @@
+# Deploying the CSI-Addons Controller
+
+The CSI-Addons Controller can be deployed by different ways:
+
+## Installation for versioned deployments
+
+The CSI-Addons Controller can also be installed  using the yaml files in `deploy/controller`.
+The versioned deployment is possible with the yaml files that get generated for the
+[latest release,](https://github.com/csi-addons/kubernetes-csi-addons/releases/latest).
```
Comment (Madhu-1) [inline docs/deploy-controller.md:9]: `latest release,` to `latest release`

File: README.md [hunk 6]
```diff
@@ -64,60 +64,13 @@ By listing the `CSIAddonsNode` CRs, the CSI-Addons Controller knows how to
...
-
-NAME                                                       DESIRED   CURRENT   READY   AGE
-replicaset.apps/csi-addons-controller-manager-687d47b8c7   1         1         1       49s
-```
+The preferred way to install the controllers is to use yaml files that are
+made available with the
+[latest release](https://github.com/csi-addons/kubernetes-csi-addons/releases/latest).
+The detailed documentation for the installation of the CSI-Addons Controller using
+the yaml file is present
+[here.](docs/deploy-controller.md) This also includes other methods to deploy the
```
Comment (Madhu-1) [inline README.md:72]: `[here.](docs/deploy-controller.md)` to `[here](docs/deploy-controller.md).`

File: docs/deploy-controller.md [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+deployment.apps/csi-addons-controller-manager created
+```
+
+* The crds.yaml create the required crds for reclaimspace operation.
+
+* The rbac.yaml creates the required rbac.
+
+* The setup-controller creates the csi-addons-controller-manager.
+
+## Installation by operator
```
Comment (Madhu-1) [inline docs/deploy-controller.md:47]: Installation by operator-sdk?

File: docs/deploy-controller.md [hunk 8]
```diff
@@ -0,0 +1,100 @@
...
+following steps:
+
+```console
+$ kubectl create namespace storage-csi-addons
+$ make operator-sdk
+$ ./bin/operator-sdk run bundle -n storage-csi-addons quay.io/csiaddons/k8s-bundle:latest
+```
+
+In the future, the bundle is expected to become available in the
+[OperatorHub][operatorhub].
```
Comment (Madhu-1) [inline docs/deploy-controller.md:59]: Link need to be added here?

File: docs/deploy-controller.md [hunk 9]
```diff
@@ -0,0 +1,100 @@
...
+clusterrolebinding.rbac.authorization.k8s.io/csi-addons-proxy-rolebinding created
+configmap/csi-addons-manager-config created
+service/csi-addons-controller-manager-metrics-service created
+
+$ kubectl create -f setup-controller.yaml
+...
+deployment.apps/csi-addons-controller-manager created
+```
+
+* The crds.yaml create the required crds for reclaimspace operation.
```
Comment (nixpanic) [inline docs/deploy-controller.md:41]: please make `filenames` quoted

**Review by nixpanic — APPROVED** [review]

File: docs/deploy-controller.md [hunk 10]
```diff
@@ -0,0 +1,100 @@
...
+$ kubectl create -f setup-controller.yaml
+...
+deployment.apps/csi-addons-controller-manager created
+```
+
+* The "crds.yaml" create the required crds for reclaimspace operation.
+
+* The "rbac.yaml" creates the required rbac.
+
+* The setup-controller creates the csi-addons-controller-manager.
```
Comment (Madhu-1) [inline docs/deploy-controller.md:45]: `setup-controller` needs to be quoted as well?

Comment (Madhu-1) [reply docs/deploy-controller.md:45]: and it should be setup-controller.yaml?

File: README.md [hunk 11]
```diff
@@ -164,7 +117,6 @@ deployment.apps/csi-addons-controller-manager created
 
 The [Contribution Guidelines](CONTRIBUTING.md) contain details on the process
 to contribute to this project.
-
```
Comment (Madhu-1) [inline README.md:167]: Is this change required?

**Review by Madhu-1 — APPROVED** [review]

File: README.md (hunk 11 above)
Comment (yati1998) [reply README.md:167]: Not required, just removed the extra space.

**Review by nixpanic — APPROVED** [review]
